### ISSUE TITLE
Remove-Item -Recurse with an -Include or -Filter argument mistakenly tries to remove the directory itself

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#16153]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Note: This issue has _partly_ been reported before, in the context of #3727, but it's worth creating a focused report.

When you combine `Remove-Item -Recurse` with either `-Include` or `-Filter`, only those items _inside_ the target directory subtree that match the pattern(s) are removed, as intended.

However, if you invoke such a command with `.` as the input path, i.e. from _inside_ the target directory, the command fails, as an attempt is seemingly made to delete the directory _itself_:

```powershell
# Switch to a temporary directory.
Push-Location -ea Stop ($tmpDir = (New-Item -Type Directory -Force (Join-Path Temp:/ $PID)).FullName)

try {

$null = New-Item -Force 1.txt, sub/2.txt

Remove-Item . -Recurse -Filter *.txt

$null = New-Item -Force 1.txt, sub/2.txt

Remove-Item . -Recurse -Include *.txt

} finally {
  # Clean up.
  Pop-Location; Remove-Item $tmpDir -Recurse
}
```

### Expected behavior

```console
Both `Remove-Item` calls should succeed and delete all `*.txt` files from the current dir's subtree.
```

### Actual behavior

```console
Both commands fail:


Cannot remove the item at '<full-path-of-current-dir>' because it is in use.
```
```


### Error details

_No response_

### Environment data

```powershell
PowerShell Core 7.2.0-preview.9
```

**Comment by 237dmitry** [comment]
> Both commands fail.

```
Remove-Item -Path .\* -Filter *.txt -Recurse -Force -ErrorAction SilentlyContinue
Remove-Item -Path .\* -Include *.txt -Recurse
``` 
Success for both commands. Try input path `.\*`
7.1.4

**Comment by mklement0** [comment]
@237dmitry, yes, that is a viable _workaround_ - but it shouldn't be necessary, and that's what this issue is about.
